File: original_scripts/r/matrix_multiplication/deepseek-coder_6.7b.r
Sure, here is the source code for matrix multiplication in R without comments and with strict programming rules adhered to:

```R
matrix_multiplication <- function(A, B) {
  if (ncol(A) != nrow(B)) stop("Matrices are not m*n and n*p")
  
  p = matrix(0, nrow(A), ncol(B))
  
  for (i in 1:nrow(A)) {
    for (j in 1:ncol(B)) {
      for (k in 1:ncol(A)) {
        p[i, j] = p[i, j] + A[i, k] * B[k, j]
      }
    }
  }
  
  return(p)
}
```
You can use this function by passing two matrices to it as arguments. For example:

```R
A = matrix(c(1,2,3,4), nrow=2)
B = matrix(c(5,6,7,8), nrow=2)
matrix_multiplication(A, B)
```

File: original_scripts/scala/matrix_multiplication/deepseek-coder_6.7b.scala
Here is the pure source code for matrix multiplication in Scala without comments and explanations:

```scala
def scalarMultiply(x: Array[Array[Double]], y: Array[Array[Double]], row1: Int, col1: Int, row2: Int, col2: Int): Double = {
  var sum = 0.0
  for (i <- 0 until row1; j <- 0 until col1) yield {
    sum += x(i)(j) * y(i)(j)
  }
  sum
}

def matrixMultiplication(a: Array[Array[Double]], b: Array[Array[Double]], r1: Int, c1: Int, r2: Int, c2: Int): Array[Array[Double]] = {
  val res = new Array[Array[Double]](r1)
  for (i <- res.indices) yield {
    res(i) = new Array[Double](c2)
  }

  if (c1 != r2) throw new Exception("Column of first matrix is not equal to row of second.")
  
  for (i <- res.indices; j <- res(0).indices) yield {
    res(i)(j) = scalarMultiply(a, b, r1, c1, r2, c2)
  }
  res
}
```
You can use these functions to multiply two matrices. Note that the indices for rows and columns are necessary to perform a multiplication operation on the given matrices. The `scalarMultiply` function calculates dot product of each row from the first matrix with corresponding row from second matrix. This result is then used in the main `matrixMultiplication` function.